pkg/datapath/loader/netlink.go

```diff
@@ -547,7 +548,7 @@ func setupGeneveDevice(dport, mtu int) error {
 //
 // Changing the port will recreate the device. Changing the MTU will modify the
 // device without recreating it.
-func setupVxlanDevice(port, mtu int) error {
+func setupVxlanDevice(port uint16, mtu int) error {
```

Comment:
Wondering why we change the sig here from int->uint since that func ends up promoting to int?
Also the const are defined as int?

Reply:
It just seemed to me that it was better to propagate the correct type (given that this represents a port) as deep as possible in the call stack. Unfortunately, the library field is an int, so we need the conversion (differently from the geneve case, which correctly accepts it as uint16). I've also updated the constants to explicitly specify the type.